src/rp2_common/pico_lwip/lwip.BUILD

```diff
@@ -2,18 +2,36 @@ load("@pico-sdk//bazel:defs.bzl", "incompatible_with_config")
 
 package(default_visibility = ["//visibility:public"])
 
+# Some of the LWIP sys_arch.h and the lwip headers depend circularly on one
+# another. Include them all in the same target.
+cc_library(
+    name = "pico_lwip_headers",
+    hdrs = glob(["**/*.h"]),
+    includes = [
+        "contrib/ports/freertos/include/arch",
+        "src/include",
+    ],
+    deps = [
+        "@pico-sdk//bazel/config:PICO_LWIP_CONFIG",
+        "@pico-sdk//src/rp2_common/pico_lwip:pico_lwip_config",
+    ],
+)
+
 cc_library(
     name = "pico_lwip_core",
     srcs = glob(["src/core/*.c"]),
-    hdrs = glob(["**/*.h"]),
-    includes = ["src/include"],
     target_compatible_with = incompatible_with_config(
         "@pico-sdk//bazel/constraint:pico_lwip_config_unset",
     ),
     deps = [
-        "@pico-sdk//bazel/config:PICO_LWIP_CONFIG",
-        "@pico-sdk//src/rp2_common/pico_lwip:pico_lwip_config",
-    ],
+        ":pico_lwip_headers",
+    ] + select({
+        "@pico-sdk//bazel/constraint:pico_freertos_unset": [
+        ],
```

Comment:
Nit: consolidate closing bracket to one line for readability:
```
"@pico-sdk//bazel/constraint:pico_freertos_unset": [],
```

Reply:
Done.